### ISSUE TITLE
npm start - npm ERR! missing script: start

### ISSUE THREAD

**Issue by jigar1859** [react-community/create-react-native-app#411]
npm ERR! missing script: start

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/...

**Comment by davidthornton** [comment]
What does your `package.json` look like?

**Comment by SDJLee** [comment]
I had the same issue. Your package,json probably wont have the **start** script. I was using the latest version of npm. Changing the version of npm to a version below 5 worked for me. I assume react-native doesn't support npm of version above 5. I installed npm version 4.6.1 and my project started successfully. I hope this helps.

**Comment by andersborgabiro** [comment]
I tried npm install -g npm@4.6.1, but still no start.

**Comment by hmake98** [comment]
I have the same issue. i try to write a code in package.json file as below
```

"scripts": {
    "start": "node your-script.js",
    "test": "echo \"Error: no test specified\" && exit 1"
  },
```
hope, it will works for you.

**Comment by nizar2yas** [comment]
same issue

**Comment by dikaiosune** [comment]
@nizar2yas @jigar1859 can you paste your package.json contents?

**Comment by endytc** [comment]
@dikaiosune  I have same issue, thisi is my package.json contents
`{
  "name": "AwesomeProject2",
  "version": "0.1.0",
  "private": true,
  "devDependencies": {
    "react-native-scripts": "1.5.0"
  }
}
`

**Comment by brentvatne** [comment]
after you install npm 4.6.1 you need to run `npm i -g create-react-native-app` again and then `create-react-native-app your-project-name`

**Comment by Marcophono2** [comment]
I am really wondering why Facebook (or the responsibles) don't explain this issue on the official "install guide" at https://facebook.github.io/react-native/docs/getting-started.html . That is something which let's me become a bit skeptical about the management behind this project. 

Best regards
Marc

**Comment by brentvatne** [comment]
@Marcophono2 - we were hoping that the issues with npm would be transient, but it looks npm 5 is still not working well. This is really frustrating for the entire JavaScript ecosystem -- should every single project now say "please don't use npm 5"? Maybe! I'll consider updating the getting started guide to suggest not using npm 5.

**Comment by andersborgabiro** [comment]
It seems like this must be a problem for Facebook as well, unless they use a bastard/fork version of npm.

**Comment by brentvatne** [comment]
It is not a problem. Facebook uses yarn, which is in many ways a better package manager than npm and also a drop-in replacement. I believe something like 30% of packages downloaded from npm are using yarn now: https://yarnpkg.com/en/

**Comment by andersborgabiro** [comment]
Sounds like time to drop npm for anything React Native related.

I'll continue with Cordova for a while longer, even though I'm close to giving it up due to the impact of code-breaking platform changes and delays in the deployment of new Cordova versions, even worse PhoneGap Build.

**Comment by brentvatne** [comment]
@andersborgabiro - if it were that easy then we would have done it. search for npm 5 on these issues and you'll see we've been patiently waiting for them to fix issues for a while.

not sure why having to install yarn or an older version of npm is a blocker for you, forcing you to stick with cordova, but to each their own! good luck

**Comment by trdevendran** [comment]
I observed one thing that is npm version based issue, npm version 4.xx is the stable one but 5.x is unstabled so we should avoid using unstable version.

Note:package.json will be configured perfectly while creating with default template if it's a stable version.

I got the below issue on the npm 5 so I downgraded to npm 4.6.1,

*******************************************************************************
ERROR: npm 5 is not supported yet
*******************************************************************************

It looks like you're using npm 5 which was recently released.

Create React Native App doesn't work with npm 5 yet, unfortunately. We
recommend using npm 4 or yarn until some bugs are resolved.

You can follow the known issues with npm 5 at:
https://github.com/npm/npm/issues/16991

*******************************************************************************
 
Devas-MacBook-Air:React-Native deva$ cd SampleTestApp/
Devas-MacBook-Air:SampleTestApp deva$ npm start
npm ERR! missing script: start

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/.npm/_logs/2018-01-28T13_26_53_649Z-debug.log

**Comment by giorgim** [comment]
@brentvatne Yeah I feel Facebook should address this issue more openly, maybe in the docs also. As this seems to be a common issue.

**Comment by projectorchid** [comment]
update "script" section in package.json  to 

"scripts": {
    "start": "webpack-dev-server --hot"
  },

**Comment by richard4s** [comment]
I have the issue as well, what's the solution?

**Comment by michaeldewolf85** [comment]
I needed to downgrade to react 4 and then start over. @richard4s @coolvasanth This should work in most cases.
```
npm install -g npm@4.6.1
npm install -g create-react-native-app
create-react-native-app AwesomeProject
npm start
```

**Comment by richard4s** [comment]
Here's what I did, and you may not like it... The reason is because your
node version and your npm versions are incompatible, some bad update. I had
to uninstall npm first then uninstall node on my computer. Then I
re-installed the latest version of node (9.5) and the latest version of
npm(5.6) and then installed yarn(1.3.2) and began using yarn to my builds.
E.g. Yarn start instead of npm start. Get back to me if you run into any
problems :)

**Comment by vyshkant** [comment]
@brentvatne The current stable version of npm is 5.6.0, and now this problem occurs on a **stable** version (5.6.0). So we either have to edit the README.md, or somehow fix this error.

IMO, this task should be reopened.

**Comment by anp** [comment]
@vyshkant that's correct, we're still waiting for upstream npm to resolve some of the issues with stable npm. We have an internal tracking issue for this and are checking back regularly. It's obviously not good when we can't support the default package manager that comes with a "default" Node install today. As far as I know, the main blocker we saw when npm5 was first released was resolved in 5.6.0. However we're still waiting on https://github.com/yarnpkg/yarn/pull/5255 to land and ship before we'll feel confident supporting npm5. @fson if I'm wrong here or there is additional detail to add, please feel free to let everyone know!

**Comment by 8Ozymandias** [comment]
@brentvatne God bless you fam! Your suggestion seems to have solved the problem so far? How did you figure this out? I swear if it wasn't for avenues like this and StackOverflow, I don't know how I would resolve certain problems especially how I am new to all this.

**Comment by 8Ozymandias** [comment]
@brentvatne It's all working good now. I am curious though, on the Getting Started page on React Native, the command they gave was  _**npm install -g create-react-native-app**_ but the command you suggested was different. Why is that they don't just provide that information on the Getting Started page. I am curious and I am thankful you provided that solution. May you be blessed by the gods, the old and the new.

**Comment by lkumar2** [comment]
I had this issue as well! I downgraded my node to 6.4.0 and my npm@4. Then created the project again which worked for me.

**Comment by magigedaniel** [comment]
Have had this issue for quite some good time. Therefore I wasted enough time. Facebook should focus on issues when raised

**Comment by xero121690** [comment]
I had this exact same issue. At least for me, it turns out i had my-first-app inside myFirstApp folder. When I used file open folder and clicked on myFirstApp, it loaded from there. However, i had the same issue when starting npm start. What i did to fix this issue is file open folder and double clicked myFirstApp folder and then clicked on my-first-app (which had all the contents in). Then it loaded perfectly fine and npm start worked again. Hopefully it helps some of you.

**Comment by diana200** [comment]
was missing the node "test", so put both nodes: start and test
and all will work :)

**Comment by rameshep1989** [comment]
I followed the instructions provided by @brentvatne and @michaeldewolf85 but i still getting the same error.



Here is the procedure i followed:

Windows PowerShell
Copyright (C) Microsoft Corporation. All rights reserved.

PS D:\Ang2\angularspree-master> npm install npm@4.6.1

> node-sass@4.8.3 install D:\Ang2\angularspree-master\node_modules\node-sass
> node scripts/install.js

Downloading binary from https://github.com/sass/node-sass/releases/download/v4.8.3/win32-x64-57_binding.node
Download complete  ] - :
Binary saved to D:\Ang2\angularspree-master\node_modules\node-sass\vendor\win32-x64-57\binding.node
Caching binary to C:\Users\Thiyagu\AppData\Roaming\npm-cache\node-sass\4.8.3\win32-x64-57_binding.node

> node-sass@4.8.3 postinstall D:\Ang2\angularspree-master\node_modules\node-sass
> node scripts/build.js

Binary found at D:\Ang2\angularspree-master\node_modules\node-sass\vendor\win32-x64-57\binding.node
Testing binary
Binary is fine
npm WARN rollback Rolling back readable-stream@2.2.9 failed (this is probably harmless): EPERM: operation not permitted, lstat 'D:\Ang2\angularspree-master\node_modules\fsevents\node_modules'
npm WARN @ngrx/core@1.2.0 requires a peer of rxjs@^5.0.0-beta.12 but none is installed. You must install peer dependencies yourself.
npm WARN cache-loader@1.2.2 requires a peer of webpack@^2.0.0 || ^3.0.0 || ^4.0.0 but none is installed. You must install peer dependencies yourself.
npm WARN codelyzer@4.2.1 requires a peer of @angular/compiler@>=2.3.1 <7.0.0 || >6.0.0-beta <7.0.0 but none is installed. You must install peer dependencies yourself.
npm WARN codelyzer@4.2.1 requires a peer of @angular/core@>=2.3.1 <7.0.0 || >6.0.0-beta <7.0.0 but none is installed. You must install peer dependencies yourself.
npm WARN codelyzer@4.2.1 requires a peer of @angular/platform-browser-dynamic@>=2.3.1 <7.0.0 || >6.0.0-beta <7.0.0 but none is installed. You
must install peer dependencies yourself.
npm WARN codelyzer@4.2.1 requires a peer of @angular/platform-browser@>=2.3.1 <7.0.0 || >6.0.0-beta <7.0.0 but none is installed. You must install peer dependencies yourself.
npm WARN codelyzer@4.2.1 requires a peer of @angular/common@>=2.3.1 <7.0.0 || >6.0.0-beta <7.0.0 but none is installed. You must install peer
dependencies yourself.
npm WARN file-loader@1.1.11 requires a peer of webpack@^2.0.0 || ^3.0.0 || ^4.0.0 but none is installed. You must install peer dependencies yourself.
npm WARN istanbul-instrumenter-loader@3.0.1 requires a peer of webpack@^2.0.0 || ^3.0.0 || ^4.0.0 but none is installed. You must install peer dependencies yourself.
npm WARN mini-css-extract-plugin@0.4.0 requires a peer of webpack@^4.4.0 but none is installed. You must install peer dependencies yourself.
npm WARN ngrx-store-freeze@0.2.2 requires a peer of @ngrx/store@>=4.0.0 but none is installed. You must install peer dependencies yourself.
npm WARN schema-utils@0.4.5 requires a peer of webpack@^2.0.0 || ^3.0.0 || ^4.0.0 but none is installed. You must install peer dependencies yourself.
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@1.1.3 (node_modules\fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.1.3: wanted {"os":"darwin","arch":"any"} (current: {"os":"win32","arch":"x64"})

+ npm@4.6.1
added 399 packages, removed 269 packages and updated 952 packages in 205.622s
PS D:\Ang2\angularspree-master> npm i -g create-react-native-app
C:\Users\Thiyagu\AppData\Roaming\npm\create-react-native-app -> C:\Users\Thiyagu\AppData\Roaming\npm\node_modules\create-react-native-app\build\index.js
+ create-react-native-app@1.0.0
added 28 packages in 9.191s
PS D:\Ang2\angularspree-master> create-react-native-app angularspree-master
Creating a new React Native app in D:\Ang2\angularspree-master\angularspree-master.

Using package manager as npm with npm interface.
Installing packages. This might take a couple minutes.
Installing react-native-scripts...

npm WARN deprecated socks@1.1.10: If using 2.x branch, please upgrade to at least 2.1.6 to avoid a serious bug with socket data flow and an import issue introduced in 2.1.0
npm notice created a lockfile as package-lock.json. You should commit this file.
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: @expo/ngrok-bin-freebsd-x64@2.2.8 (node_modules\@expo\ngrok-bin-freebsd-x64):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for @expo/ngrok-bin-freebsd-x64@2.2.8: wanted {"os":"freebsd","arch":"x64"} (current: {"os":"win32","arch":"x64"})
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: @expo/ngrok-bin-darwin-x64@2.2.8 (node_modules\@expo\ngrok-bin-darwin-x64):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for @expo/ngrok-bin-darwin-x64@2.2.8: wanted {"os":"darwin","arch":"x64"} (current: {"os":"win32","arch":"x64"})
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: @expo/ngrok-bin-darwin-ia32@2.2.8 (node_modules\@expo\ngrok-bin-darwin-ia32):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for @expo/ngrok-bin-darwin-ia32@2.2.8: wanted {"os":"darwin","arch":"ia32"} (current: {"os":"win32","arch":"x64"})
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: @expo/ngrok-bin-freebsd-ia32@2.2.8 (node_modules\@expo\ngrok-bin-freebsd-ia32):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for @expo/ngrok-bin-freebsd-ia32@2.2.8: wanted {"os":"freebsd","arch":"ia32"} (current: {"os":"win32","arch":"x64"})
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: @expo/ngrok-bin-linux-arm64@2.2.8 (node_modules\@expo\ngrok-bin-linux-arm64):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for @expo/ngrok-bin-linux-arm64@2.2.8: wanted {"os":"linux","arch":"arm64"} (current: {"os":"win32","arch":"x64"})
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: @expo/ngrok-bin-linux-arm@2.2.8 (node_modules\@expo\ngrok-bin-linux-arm):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for @expo/ngrok-bin-linux-arm@2.2.8: wanted {"os":"linux","arch":"arm"} (current: {"os":"win32","arch":"x64"})
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: @expo/ngrok-bin-linux-ia32@2.2.8 (node_modules\@expo\ngrok-bin-linux-ia32):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for @expo/ngrok-bin-linux-ia32@2.2.8: wanted {"os":"linux","arch":"ia32"} (current: {"os":"win32","arch":"x64"})
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: @expo/ngrok-bin-linux-x64@2.2.8 (node_modules\@expo\ngrok-bin-linux-x64):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for @expo/ngrok-bin-linux-x64@2.2.8: wanted {"os":"linux","arch":"x64"} (current: {"os":"win32","arch":"x64"})
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: @expo/ngrok-bin-sunos-x64@2.2.8 (node_modules\@expo\ngrok-bin-sunos-x64):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for @expo/ngrok-bin-sunos-x64@2.2.8: wanted {"os":"sunos","arch":"x64"} (current: {"os":"win32","arch":"x64"})
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: @expo/ngrok-bin-win32-ia32@2.2.8-beta.1 (node_modules\@expo\ngrok-bin-win32-ia32):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for @expo/ngrok-bin-win32-ia32@2.2.8-beta.1: wanted {"os":"win32","arch":"ia32"} (current: {"os":"win32","arch":"x64"})

+ react-native-scripts@1.14.0
added 442 packages in 117.254s
Installing dependencies using yarn...

'yarnpkg' is not recognized as an internal or external command,
operable program or batch file.
events.js:183
      throw er; // Unhandled 'error' event
      ^

Error: spawn yarnpkg ENOENT
    at notFoundError (D:\Ang2\angularspree-master\angularspree-master\node_modules\cross-spawn\lib\enoent.js:11:11)
    at verifyENOENT (D:\Ang2\angularspree-master\angularspree-master\node_modules\cross-spawn\lib\enoent.js:46:16)
    at ChildProcess.cp.emit (D:\Ang2\angularspree-master\angularspree-master\node_modules\cross-spawn\lib\enoent.js:33:19)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:198:12)
PS D:\Ang2\angularspree-master> npm start
npm ERR! missing script: start

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\Thiyagu\AppData\Roaming\npm-cache\_logs\2018-07-16T04_37_28_137Z-debug.log
PS D:\Ang2\angularspree-master>

**Comment by richard4s** [comment]
Here's what I did, and you may not like it... The reason is because your
node version and your npm versions are incompatible, some bad update. I had
to uninstall npm first then uninstall node on my computer. Then I
re-installed the latest version of node (9.5) and the latest version of
npm(5.6) and then installed yarn(1.3.2) and began using yarn to my builds.
E.g. Yarn start instead of npm start. Get back to me if you run into any
problems :)

**Comment by Alisa-1117** [comment]
@hmake98 hi, I added that two lines code in packjson, but it not still start, what should i do?

**Comment by ezerbelhassen** [comment]
maybe you forgot to run npm install

**Comment by ankita1427** [comment]
what is : your-script.js

**Comment by azizahmedkhan** [comment]
uninstalling and reinstalling create-react-app solved the problem for me.

**Comment by ananya09patankar** [comment]
> what is : your-script.js

same doubt

**Comment by brentvatne** [comment]
this issue is very old and the project has been completely rewritten since then. if you have some related sounding problem, please create a new issue with more info, thanks!
